sysroot_creator/scripts/create_cc_sysroot.py

```diff
@@ -129,52 +130,69 @@ def build_workspace_sysroot_image(
         platform: Platform,
         docker_args: DockerConfig,
         image_tag: str):
-    log('Fetching sysroot base image {}'.format(docker_args.base_image))
+    logger.info('Fetching sysroot base image: {}'.format(docker_args.base_image))
     DOCKER_CLIENT.images.pull(docker_args.base_image)
     # FIXME consider moving constants to static fields
     workspace_dockerfile_path = (Path(__file__).parent / SYSROOT_DIR_NAME /
                                  DOCKER_WS_NAME)
 
     buildargs = {
         'ROS2_BASE_IMG': docker_args.base_image,
-        'ROS2_WORKSPACE': './' + SYSROOT_DIR_NAME,
+        'ROS2_WORKSPACE': './{}'.format(SYSROOT_DIR_NAME),
         'ROS_DISTRO': platform.distro,
         'TARGET_TRIPLE': platform.cc_toolchain
     }
-    log('Building workspace image {} (this might take some time...)'
-        .format(image_tag))
-    # FIXME: this gives no output until the whole image is built,
-    # at least implement some spinner
+    logger.info('Building workspace image: {}'.format(image_tag))
+
     try:
-        workspace_image, ws_image_build_logs = DOCKER_CLIENT.images.build(
+        # Switch to low-level API to expose build logs
+        docker_client = docker.APIClient(base_url='unix://var/run/docker.sock')
+        log_generator = docker_client.build(
             path='.',
             dockerfile=str(workspace_dockerfile_path),
             tag=image_tag,
             buildargs=buildargs,
             quiet=False,
             nocache=docker_args.nocache,
-            network_mode=docker_args.network_mode)
-        print('Docker build output')
-        print_docker_logs(ws_image_build_logs)
-        print()
+            network_mode=docker_args.network_mode,
+            decode=True)
+        # Progress bar with postfix
+        with tqdm(log_generator) as t:
+            for chunk in t:
+                try:
+                    # Remove carriage returns and new lines ('\r\n')
```

Comment:
Seems like this could be a separate utility function.
Do we have unit tests for those? Making those top-level utility functions would help us making sure all this string manipulation code works well.